### PR TITLE
feat(deployment): support custom CAs

### PR DESCRIPTION
**Pull request by migmartri**
Add support to providing custom CAs available during TLS verification by the application HTTP clients.

Example of applying this change and adding a custom CA

- Will create a secret
- Will mount it in a default system path understood by golang's crypto library https://go.dev/src/crypto/x509/root_linux.go

I've tested it locally pointing my OIDC server to an nginx with a TLS certificate provided by the [staging certificate](https://letsencrypt.org/docs/staging-environment/) from let's encrypt. 

Closes https://github.com/chainloop-dev/chainloop/issues/956

```diff
 ---
+# Source: chainloop/templates/controlplane/customcas.secret.yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: chainloop-2-controlplane-custom-cas
+  labels:
+    app.kubernetes.io/name: chainloop
+    helm.sh/chart: chainloop-1.61.0
+    app.kubernetes.io/instance: chainloop-2
+    app.kubernetes.io/managed-by: Helm
+    app.kubernetes.io/version: "v0.91.6"
+    app.kubernetes.io/part-of: chainloop
+    app.kubernetes.io/component: controlplane
+data:
+  custom-0.crt: "LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUZtRENDQTRDZ0F3SUJBZ0lRVTlDODduTXBPSUZLWXBmdk9IRkhGREFOQmdrcWhraUc5dzBCQVFzRkFEQm0KTVFzd0NRWURWUVFHRXdKVlV6RXpNREVHQTFVRUNoTXFLRk5VUVVkSlRrY3BJRWx1ZEdWeWJtVjBJRk5sWTNWeQphWFI1SUZKbGMyVmhjbU5vSUVkeWIzVndNU0l3SUFZRFZRUURFeGtvVTFSQlIwbE9SeWtnVUhKbGRHVnVaQ0JRClpXRnlJRmd4TUI0WERURTFNRFl3TkRFeE1EUXpPRm9YRFRNMU1EWXdOREV4TURRek9Gb3daakVMTUFrR0ExVUUKQmhNQ1ZWTXhNekF4QmdOVkJBb1RLaWhUVkVGSFNVNUhLU0JKYm5SbGNtNWxkQ0JUWldOMWNtbDBlU0JTWlhObApZWEpqYUNCSGNtOTFjREVpTUNBR0ExVUVBeE1aS0ZOVVFVZEpUa2NwSUZCeVpYUmxibVFnVUdWaGNpQllNVENDCkFpSXdEUVlKS29aSWh2Y05BUUVCQlFBRGdnSVBBRENDQWdvQ2dnSUJBTGJhZ0VkRFRhMVFnR0JXU1lreU1oc2MKWlhFTk9CYVZSVE1YMWhjZUpFTmdzTDBNYTQ5RDNNaWxJNEtTMzhtdGttZEY2Y1BXbkwrK2ZnZWhUMEZiUkhaZwpqT0VyOFVBTjRqSDZvbWpyYlREKytWWm5lVHNNVmFHYW1RbURkRmw1ZzFnWWFpZ2trbXg4T2lDTzY4YTRRWGc0CndTeW42aURpcEtQOHV0c0UreDFFMjhTQTc1SE9ZcXBkcms0SEd4dVVMdmxyMDN3WkdUSWYvb1J0Mi9jK2RZbUQKb2FKaGdlK0dPckxBRVFCeU83KzgrdnpPd3BOQVBFeDZMVytjckVFWjdlQlhpaDZWUDE5c1RHeTN5ZnFLNXRQdApUZFhYQ09RTUtBcCtnQ2ovVkJ5aG1JciswaU5EQzU0MGd0dlYzMDNXcGNid25ra0xZQzBGdDJjWVV5SHRrc3RPCmZSY1JPK0syY1pvem9Td1ZQeUI4L0o5UnBjUkszamduWDlsdWpmd0EvcEFiUDBKMlVQUUZ4bVdGUlFuRmphcTYKcmtxYk5FQmdMeStrRkwxTkVzUmJ2RmJLclJpNWJZeTJsTm1zMk5KUFp2ZE5RYlQvMmRCWkttSnF4SGt4Q3VPUQpGamhKUU5lTytOam0xWjFpQVRTLzNydHMyeVpscVhLc3hRVXpONnZOYkQ4S25YUk1FZU9YVVl2YlY0bHFmQ2Y4Cm1TMTRXRWJTaU15ODdHQjVTOXVjU1YxWFVybFRHNVVHY01TWk9CY0VVcGlzUlBFbVFXVU9UV0lvRFE1Rk9pYS8KR0krS2k1MjNyMnJ1RW1ibUczN0VCU0JYZHhJZG5kcXJqeStRVkFtQ2VieUR4OWVWRUdPSXBuMjZiVzVMS2VydQptSnhhL0NGQmFLaTRiUnZtZEpSTEFnTUJBQUdqUWpCQU1BNEdBMVVkRHdFQi93UUVBd0lCQmpBUEJnTlZIUk1CCkFmOEVCVEFEQVFIL01CMEdBMVVkRGdRV0JCUzE4Mlh5L3JBS2toLzdQSDN6UktDc1l5WERGREFOQmdrcWhraUcKOXcwQkFRc0ZBQU9DQWdFQW5jRFpOeXREYnJyVmU2OFVUNnB5MWxmRjJoNlRtMnA4cm80Mmk4N1dXeVAyTEs4WQpuTEhDMGh2TmZXZVdtalpRWUJRZkdDNWM3YVFSZXphayt0SExkbXJOS0hrbjVrbis5RTlMQ2pDYUVzeUlJbjJqCnFkSGxBa2VwdS9DM0tuTnRWeDV0VzA3ZTVidklqSlNjd2tDRGJQM2FrV1FpeFBwUkZBc25QK1VMeDdrMGFPMXgKcUFlYUFoUTJyZ28xRjU4aGNmbGdxS1RYbnBQTTAyaW50VmZpVlZrWDVHWHBKaks1RW9RdExjZXlHT3JreGxNLwpzVFBxNFVybnlwbXNxU2FnV1YzSGNVbFl0RGluYytudWtGazZlUjRYa3pYQmJ3S2FqbDBZanp0ZnJDSUhPbjVRCkNKTDZURVJWRGJNL2FBUGx5OGtKMXNXR0x1dnZXWXpNWWdMekR1bC8vclVGMTBnRU1XYVhWWlY1MUtwUzlEWS8KNUN1bnV2Q1htRVFKSG83a0djVmlUN3NFVG42Sno5S09odlljWGtKN3BvNmQ5M0Evank0R0tQSVBuc0tLTkVtUgp4VXVYWTR4UmRoNDV0TUpuTFRVRGRDOUZJVTBmbFRlTzkvdk5wVkE4T1BVMWkxNHZDeitNVThLWDFiVjNHWG0vCmZ4bEI3VkJCalg5djVvVWVwMG8vajY4Ui9pRGxDT000VlZmUmE4Z1g2VDJGVTdmTmRhdHZHcm83dVF6SXZXb2YKZ045V1V3Q2JFTUJ5L1loQlNyWHljS0E4Y3JnR2czeDFtSXNvcG44OEpLd21NQmE2OG9TN0VITTl3N0M0eTcxTQo3RGlBKy85UWRwOVJCV0pwVFM5aS9tRG5KZzF4dm84WHo0OW1ycmdmbWNBWFRDSnFYaTI0TmF0STNPYz0KLS0tLS1FTkQgQ0VSVElGSUNBVEUtLS0tLQ=="
+---
 # Source: chainloop/templates/controlplane/jwt_cas_private_key.secret.yaml
 apiVersion: v1
 kind: Secret
@@ -1430,7 +1446,7 @@
     metadata:
       annotations:
         checksum/config: daa5fd5324f11e1802d549215a36da1e567627536c8a0658af8a13ac8ec2146d
-        checksum/secret-config: 88fdc4664ce52cc1b4928c92a4903e96ce7462dffc3fe6c6ed41164540a079aa
+        checksum/secret-config: 49be03a7a7d243ca88ad971ac5958ada0b40ce3596251eff6eb8a3ae936f6bc3
         checksum/cas-private-key: 7bc159cd8a76c55ccef7fa39b7780459d289d515eade0115b574cc1c2aa27555
         kubectl.kubernetes.io/default-container: controlplane
       labels:
@@ -1492,6 +1508,11 @@
               mountPath: /tmp
             - name: jwt-cas-private-key
               mountPath: /secrets
+            - name: custom-cas
+              # NOTE: /etc/ssl/certs already contains the system CA certs
+              # Let's use another known path https://go.dev/src/crypto/x509/root_linux.go
+              mountPath: /etc/pki/tls/certs
+              readOnly: true
       volumes:
         - name: config
           projected:
@@ -1500,6 +1521,11 @@
                 name: chainloop-2-controlplane
             - configMap:
                name: chainloop-2-controlplane
+        - name: custom-cas
+          projected:
+            sources:
+            - secret:
+               name: chainloop-2-controlplane-custom-cas
         # required for the plugins to store the socket files
         - name: tmp
           emptyDir: {}

```

### CODE REVIEW

File: deployment/chainloop/templates/controlplane/deployment.yaml [hunk 1]
```diff
@@ -91,6 +91,13 @@ spec:
             - name: gcp-secretmanager-serviceaccountkey
               mountPath: /gcp-secrets
             {{- end }}
+            {{- if (not (empty .Values.controlplane.customCAs)) }}
+            - name: custom-cas
+              # NOTE: /etc/ssl/certs already contains the system CA certs
+              # Let's use another known path https://go.dev/src/crypto/x509/root_linux.go
+              mountPath: /etc/pki/tls/certs
```
Comment (jiparis) [inline deployment/chainloop/templates/controlplane/deployment.yaml:98]: I think you can still mount it in the standard /etc/ssl/certs folder. That's what I've done previously in similar changes. Other tools or libraries might not support this other path.

Comment (javirln) [reply deployment/chainloop/templates/controlplane/deployment.yaml:98]: A subpath within `/etc/ssl/certs/` won't work right?

**Review by javirln — APPROVED** [review]
Added a small comment.

File: deployment/chainloop/templates/controlplane/deployment.yaml (hunk 1 above)
Comment (migmartri) [reply deployment/chainloop/templates/controlplane/deployment.yaml:98]: It doesn't seem to load recursively no :(

https://cs.opensource.google/go/go/+/master:src/crypto/x509/root_unix.go;l=62;drc=fe36ce669c1a452d2b0e81108a7e07674b50692a

Comment (jiparis) [reply deployment/chainloop/templates/controlplane/deployment.yaml:98]: As long as the entire folder doesn't get overridden, you can have as many files as you want there. Just make sure of mounting the file, not the whole folder.

Comment (migmartri) [reply deployment/chainloop/templates/controlplane/deployment.yaml:98]: Right, but the files are dynamic, so you can add as many certs as you want. I could definitely generate dynamically the mount points indeed. 

I'll do it in another patch

Comment (migmartri) [reply deployment/chainloop/templates/controlplane/deployment.yaml:98]: Tracked https://github.com/chainloop-dev/chainloop/issues/965

**Pull request merged**